src/System.Private.Reflection.Core/src/System/Reflection/Runtime/General/QSignatureTypeHandle.NativeFormat.cs

```diff
@@ -0,0 +1,36 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+// Collection of "qualified handle" tuples.
+//
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+using Internal.Metadata.NativeFormat;
+using Internal.Runtime.TypeLoader;
+
+namespace System.Reflection.Runtime.General
+{
+    public partial struct QSignatureTypeHandle
```

Comment:
You've got a QMethodDefinition that a native-specific NativeFormatHandle property - I think the same pattern can be followed by QTypeDefRefOrSpec. I don't like the idea of having both QTypeDefRefOrSpec and QSignatureType around - they're modeling the same concept even if the underlying representation schema is different on ECMA.